applications/dot_marketplace-phase2.md

```diff
@@ -0,0 +1,218 @@
+| --- | --- | --- |
+| 0a | License | Apache 2.0 |
+| 0b | Documentation | We will provide both inline documentation of the code and a tutorial that explains how a user can use DOT Marketplace and understand the flow of tasking pallet. |
+| 0c | Testing Guide | Functions will be covered by unit tests, the documentation will describe how to run these tests. We will also provide scripts to help deploy, run and test the build. |
+| 0d | Docker Image | Docker image of the build |
+| 1 | Migration from frame v1 to frame v2 | The [existing application backend](https://github.com/WowLabz/tasking_backend/blob/Phase1_Milestone3/pallets/pallet-tasking/src/lib.rs) will be migrated to frame v2 to account for the new features provided by the framework |
+| 2 | Chat Pallet | Chat functionality is to be exposed and consumed between the customer and the service provider to ease communication and this will be integrated with the tasking pallet's frame v2 version |
+|  |  |  |
+
+### **Milestone 2**
+
+* **Estimated duration:** 2 weeks
+* **FTE:**  2
+* **PTE:**  2
+* **Costs:**  15,900 USD  
+
+
+In continuation to previous work, this milestone involves creation of a on chain decentralized court to handle dispute resolution. Being a juror is one of the user incentives that can be achieved thanks to the rating module mentioned in the first phase of dot marketplace. The entire milestone covers the Rust/Substrate code implementation.
+
+| Sr no. | Deliverable | Description |
+| --- | --- | --- |
+| 0a | License | Apache 2.0 |
+| 0b | Documentation | We will provide both inline documentation of the code and a tutorial that explains how a user can use DOT Marketplace and understand the flow of tasking pallet. |
+| 0c | Testing Guide | Functions will be covered by unit tests, the documentation will describe how to run these tests. We will also provide scripts to help deploy, run and test the build. |
+| 0d | Docker Image | Docker image of the build |
+| 1 | Decentralized Court Module | An on chain decentralized court for dispute resolution within the ecosystem. |
+| 1a | Disapprove Task  | In the case of a customer not being satisfied by the work submitted by the service provider (worker). A set of jurors is formed (court-summon) to resolve the dispute and pass a verdict. |
+| 1b | Disapprove Rating | The customer or the service provider, once they have received their rating for a particular task and are not satisfied by it. |
+| 1c | General Dispute | A general dispute function for cases that do not fall under the categories mentioned in 1a and 1b. |
+| 2 | Jury | The chain specification of the testnet is modified to include more users with necessary specifications to be a part of the jury. The specifications include having average user rating above a certain threshold and being an expert in the field of the task. A list of potential jurors are notified and they have a period of one day to accept jury duty, with the maximum number of juors capped to 5 per dispute. |
```

Comment:
> having average user rating above a certain threshold

1. Will this threshold be configurable or hardcoded? If configurable, how can it be changed?
2. How many jurors have to be elected for settling a dispute? Is it configurable? If configurable, how can it be changed?
3. What happens if there are not enough jurors that decide to become a juror within the 3d time period?
4. What happens if there are not enough jurors that decide to cast their vote?

Reply:
1. The average threshold at the moment (i.e. 4) is hardcoded. We can provide a functionality for setting the criteria for average rating. And this can be set when raising the dispute.
2. Atleast 1 juror is required to settle the dispute. The max cap of jurors accepting jury duty is set, but it also can be added as a configuration when registering a dispute.
3. The program will do the calculations from among the jurors who participated
4. Same as 3

Reply:
Ok thx. Also here, I think it'd be a good idea to integrate this information into the application document.

Reply:
@takahser Updated the application document. Please check and let me know.

Reply:
In the description of Milestone 2 - [general dispute and court schedule requirements](https://github.com/w3f/Grants-Program/blob/78d842b1d9fd11aaed20447eaa1015033a7b2056/applications/dot_marketplace-phase2.md#milestone-2)